torch/_inductor/codegen/wrapper.py

```diff
@@ -33,14 +33,22 @@
 def buffer_reuse_key(node: ir.Buffer):
     size = node.get_size()
     stride = node.get_stride()
-    last_element = sympy_dot([s - 1 for s in size], stride)
-    return (
-        node.get_device(),
-        node.get_dtype(),
-        V.graph.sizevars.simplify(sympy_product(size)),
-        # Detect gaps in tensor storage caused by strides
-        V.graph.sizevars.size_hint(last_element),
-    )
+    if any([str(s) in V.graph.data_dep_symbol_to_expr_str for s in size]):
+        return (
+            node.get_device(),
+            node.get_dtype(),
+            str(size),
+            str(stride),
```

Comment:
@jansel says to use `sympy_str` here instead of str, it will be faster. Also, recording the entire stride is not necessary, you should use the last element calculation. Actually, I would suggest NOT having the top level conditional here; instead, you want to add a `handle_unbacked` call around the fourth tuple element, which tests if last_element is size hint'able, and if it isn't switches to `sympy_str`. (Actually, it's kind of weird that line 40 can take sympy symbols, I wonder what's up with that...)

Reply:
Found that this is no longer needed after switching to ExternKernelAlloc